backend/lib/src/services/msp.rs

```diff
@@ -363,7 +361,14 @@ impl MspService {
 
                     // Convert u128 price to BigDecimal and multiply
                     let price_bd = BigDecimal::from(current_price_per_giga_unit_per_tick);
-                    let cost = amount_provided * &unit_to_giga_unit * price_bd;
+
+                    // Matches the computation done in the runtime
+                    //
+                    // (price * amount) / gigaunit
+                    let cost = (price_bd * amount_provided) / shp_constants::GIGAUNIT;
+
+                    // we truncate the decimal digits of the cost per tick
+                    let cost = cost.with_scale_round(0, bigdecimal::RoundingMode::Down);
```

Comment:
```suggestion
                    // Truncate the decimal digits of the cost per tick
                    let cost = cost.with_scale_round(0, RoundingMode::Down);
```
And import `RoundingMode` at the top of the file 😜